client/state/selectors/get-site-default-post-format.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * External
+ */
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getRawSite, getSiteOption } from 'state/sites/selectors';
+import { getSiteSettings } from 'state/site-settings/selectors';
+
+/**
+ * Returns the default post format of a site.
+ * Returns null if the site is unknown.
+ *
+ * @param  {Object}    state   Global state tree
+ * @param  {Number}    siteId  The ID of the site we're querying
+ * @return {?String}           The default post format of that site
+ */
+export default function getSiteDefaultPostFormat( state, siteId ) {
+	const site = getRawSite( state, siteId );
+	if ( ! site ) {
+		return null;
+	}
+
+	let defaultPostFormat = get( getSiteSettings( state, siteId ), 'default_post_format' );
+	if ( ! defaultPostFormat ) {
+		defaultPostFormat = getSiteOption( state, siteId, 'default_post_format' );
+	}
+
+	if ( ! defaultPostFormat || defaultPostFormat === '0' ) {
```

Comment:
> Settings and site data are separately retrieved from the API, so it's not enough to check just that the raw site data is available

Yes, but since we're now migrating away from `sites-list`, the `settings` will not be available in the `site` object. And that is why we're falling back to the setting in the site object. Also, since we're using the Redux site in these cases `.settings` is not there anyway, so we're only improving the current solution. I'm open to any better alternatives.

> I'm not sure if we want to assume a "standard" format in the case that we don't know what the site or site settings are; typically we'd return null from a selector in those cases to indicate a "can't know" state.

For me, the `getSiteDefaultPostFormat` selector is a more specific one, because I'd expect it to return a default value that we can actually use. In the WordPress core, empty string is saved for the post format of a post when "standard" is selected, and if you've never touched post formats in your installation, the "Standard" is the default post format when inserting a new post. So, it would make sense for me that we'd reach the "can't know" state only if we don't have information about the site yet. But if we have information about the site, it would make sense to return the `standard` post format as the default post format if there's no other default post format specified. In other words, standard is the "default" default post format in WordPress, not `null`.